packages/atomic/src/utils/init-queue.spec.ts

```diff
@@ -0,0 +1,69 @@
+import {describe, it, expect, beforeEach, vi} from 'vitest';
+import {
+  markParentAsReady,
+  isParentReady,
+  queueEventForParent,
+} from './init-queue';
+import {InitializeEvent} from './init-queue';
+
+describe('init-queue', () => {
+  let parent: HTMLElement;
+  let child: HTMLElement;
+  let event: InitializeEvent;
+
+  beforeEach(() => {
+    parent = document.createElement('div');
+    child = document.createElement('div');
+    event = new CustomEvent('initialize', {
+      detail: () => {},
+    }) as InitializeEvent;
+    document.body.appendChild(parent);
+    document.body.appendChild(child);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(parent);
+    document.body.removeChild(child);
+  });
+
+  it('should mark parent as ready', () => {
```

Comment:
The tests do not verify the behavior when `markParentAsReady` is called multiple times for the same parent.
```ts
it('should handle multiple calls to markParentAsReady gracefully', () => {
  markParentAsReady(parent);
  expect(isParentReady(parent)).toBe(true);
  markParentAsReady(parent); // Call again
  expect(isParentReady(parent)).toBe(true); // Should still be true
});
```
Subsequent calls should not behave differently.